packages/mcp/src/index.ts

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env node
+import { StdioServerTransport } from "@modelcontextprotocol/sdk/server/stdio.js";
+import chalk from "chalk";
+
+import { RecallAgentToolkit } from "@recallnet/agent-toolkit/mcp";
+import {
+  Configuration,
+  Permission,
+  Resource,
+} from "@recallnet/agent-toolkit/shared";
+
+/**
+ * Options for the Recall MCP server
+ * @param tools - The tools to enable
+ * @param privateKey - The private key to use
+ * @param network - The Recall network to use: `testnet` or `localnet` (defaults to `testnet`)
+ */
+type Options = {
+  tools?: string[];
+  privateKey?: string;
+  network?: string;
+};
+
+// The server will accept flags for a `--private-key` (hex string), `--tools` (comma separated),
+// and `--network` (e.g., `testnet` or `localnet`)
+const ACCEPTED_ARGS = ["private-key", "tools", "network"];
+
+// Note: since we require a private key, we only expose both read and write tools
+const ACCEPTED_TOOLS = [
+  "account.read",
+  "account.write",
+  "bucket.read",
+  "bucket.write",
+];
+
+// eslint-disable-next-line turbo/no-undeclared-env-vars
+const { RECALL_PRIVATE_KEY, RECALL_NETWORK } = process.env;
+
+/**
+ * Parse the command line arguments
+ * @param args - The command line arguments—one or more of:
+ *  `--tools=all` (or `--tools=resource.action`, like `--tools=account.read`),
+ *  `--private-key=$KEY`, or `--network=$NETWORK`
+ * @returns The parsed options
+ */
+export function parseArgs(args: string[]): Options {
```

Comment:
I don't _love_ that we have a custom parseArgs function here. There _must_ be small libraries that we could use here instead? I guess probs fine so long as it's easy to maintain.

Reply:
i'll tal. i didnt wanna change too much from the example MCP impl, but might as well do it right.